File: backend/app/schemas/order.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import List, Annotated
from app.core.enums import OrderAction, OrderStatus

class OrderItemCreate(BaseModel):
    model_id: int
    quantity: int
    unit_price: int
    

class OrderCreate(BaseModel):
    shipping_address: str
    items: Annotated[List[OrderItemCreate], Field(min_length=1)]
    

class OrderUpdate(BaseModel): 
    pass    


class OrderItemOut(BaseModel):
    model_id: int
    quantity: int
    unit_price: int
    
    class Config: 
        from_attributes = True
        
class OrderOut(BaseModel):
    id: int
    user_id: int
    shipping_address: str
    order_date: datetime
    status: OrderStatus
    items: List[OrderItemOut]
    
    class Config:
        from_attributes = True
        

class OrderTransitionRequest(BaseModel):
    action: OrderAction

File: frontend/src/types/order.ts
export type OrderStatus =
    | "pending"
    | "confirmed"
    | "delivering"
    | "delivered"
    | "cancelled"
    | "refunded";

export type OrderAction =
    | "admin_confirm"
    | "admin_ship"
    | "user_confirm_delivery"
    | "cancel";

export interface OrderItemCreate {
    model_id: number;
    quantity: number;
    unit_price: number;
}

export interface OrderCreate {
    shipping_address: string;
    items: OrderItemCreate[];
}

export interface OrderItemOut {
    model_id: number;
    quantity: number;
    unit_price: number;
}

export interface OrderOut {
    id: number;
    user_id: number;
    shipping_address: string;
    order_date: string;
    status: OrderStatus;
    items: OrderItemOut[];
}

export interface CancelOrderRequest {
    order_id: number;
}

export interface CancelOrderResponse {
    status: string;
    message?: string | null;
}

export interface OrderTransitionRequest {
    action: OrderAction;
}
